### DISCUSSION TITLE
[api-gateways] Back Office not working with Envoy

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#106] [Q&A] [answered]
Hi guys,
we changed the `nginx` api-gateway with the `envoy` api-gateway and the back office site stopped working.

Is there some configuration to add to `envoy` to support the back office?

I have the following errors inside the inspector console, that I have not when using `nginx`:
<img width="1919" alt="envoy-back-office-error" src="https://user-images.githubusercontent.com/28152586/203098759-512da502-94a6-410a-a17c-89eedb562413.png">

**Comment by davidebianchi** [comment, marked as answer]
Hi @silversoul93, why the mime type is text/html? It seems to be css or js

**Reply by frengooo** [reply]
I don't know, the backoffice was configured through the application (not by me) and changing the api-gateway it doesn't work anymore.
I tried to delete the `envoy` gw and add the `nginx` one again and it worked again, but we need to use `envoy` for that project.

**Reply by frengooo** [reply]
@davidebianchi maybe `envoy` overrides the content-type of the static file?
<img width="150" alt="Screenshot 2022-11-22 alle 10 26 35" src="https://user-images.githubusercontent.com/28152586/203276979-fb5a2670-a496-448d-8717-c4c380fd951f.png">

**Reply by frengooo** [reply]
Probably is related to `micro-lc`, because the client's website is correctly served through `envoy`

**Reply by frengooo** [reply]
We found the root cause with the help of @davidebianchi.
The problem is a bug on `micro-lc`, not a envoy issue.
I already opened a bug on the tool:
https://github.com/micro-lc/micro-lc/issues/544

**Reply by davidebianchi** [reply]
As a final check, the api-gateway based on nginx change the content-type automatically (adding the default types directive) if it ends with `.js`. So, the nginx based api-gateway hide the wrong content-type header but this is not do by envoy.
